### PR TITLE
minor documentation changes

### PR DESCRIPTION
**Pull request by herbertgithubaccount**
i reworded the documentation in a few places, added some more detail here and there, and fixed/tweaked the function parameters/return types for some functions.

i'd be happy to undo some changes (or revise them) if people would like

the documentation was also rebuilt

### notes:
* i changed the return type of `tes3.getLuaModMetadata` to be `MWSE.Metadata`. however, this is only defined in `core/lib/dependencyManagement/DependencyManager.lua`, so there's no link to it in the generated `mkdocs` documentation. but it's still useful for code completion in visual studio.
  * i can add the `MWSE.Metadata` class to the `autocomplete` folder if people think that would be appropriate

### CODE REVIEW

File: autocomplete/definitions/global/table/filter.lua [hunk 1]
```diff
@@ -8,7 +8,7 @@ Any additional arguments will be passed to `f`. For example, `table.filter(t, f,
 ]],
 	arguments = {
 		{ name = "t", type = "table" },
-		{ name = "f", type = "fun(k: unknown, v: unknown, ...)", "The function to use when filtering values of `t`. (This is sometimes called a predicate function.)" },
+		{ name = "f", type = "fun(k: unknown, v: unknown, ...): boolean|unknown", "The function to use when filtering values of `t`. (This is sometimes called a predicate function.)" },
```
Comment (C3pa) [inline autocomplete/definitions/global/table/filter.lua:11]: It's not enforced in the implementation, but let's limit the return value to `boolean`.

File: autocomplete/definitions/global/table/filterarray.lua [hunk 2]
```diff
@@ -7,7 +7,7 @@ Any additional arguments will be passed to `f`. For example, `table.filterarray(
 When an element gets filtered out, the index of subsequent items will be shifted down, so that the resulting table plays nicely with the `#` operator and the `ipairs` function.]],
 	arguments = {
 		{ name = "arr", type = "table" },
-		{ name = "f", type = "fun(i: integer, v: unknown, ...)", "The function to use when filtering values of `t`. (This is sometimes called a predicate function.)" },
+		{ name = "f", type = "fun(i: integer, v: unknown, ...): boolean|unknown", "The function to use when filtering values of `t`. (This is sometimes called a predicate function.)" },
```
Comment (C3pa) [inline autocomplete/definitions/global/table/filterarray.lua:10]: Same as above.

File: autocomplete/definitions/global/json/encode.lua [hunk 3]
```diff
@@ -1,9 +1,9 @@
 return {
 	type = "function",
-	description = [[Create a string representing the object. Object can be a table, a string, a number, a boolean, nil, json.null or any object with a function __tojson in its metatable. A table can only use strings and numbers as keys and its values have to be valid objects as well. It raises an error for any invalid data types or reference cycles.
+	description = [[Create a string representing the object. Object can be a `table`, `string`, `number`, `boolean`, `nil`, `json.null`,  or any object with a `__tojson` function in its `metatable`. A `table` can only use strings and numbers as keys, and its values have to be valid objects as well. This function will raise an error if called on an invalid data type or on a data struture that contains reference cycles.
```
Comment (C3pa) [inline autocomplete/definitions/global/json/encode.lua:3]: on a data _structure_

File: autocomplete/definitions/global/json/decode.lua [hunk 4]
```diff
@@ -2,8 +2,9 @@ return {
 	type = "function",
 	description = [[Decode string into a table.
 
-!!! warning
-	If the table encoded as json had both string and integer indices, this process converted all the integer indices to strings. For example, `[1]` was converted to `["1"]`. So, when loading mixed tables from json, this needs to be considered.
+!!! warning "json does not support `number` indices"
+	As a result, the `table` returned by this function won't have any `number` indices. (e.g., the first key of an array-style table will be decoded as `["1"]`, not as `[1`].)
```
Comment (C3pa) [inline autocomplete/definitions/global/json/decode.lua:6]:
```
`[1`] -> `[1]`
```

File: autocomplete/definitions/global/string/split.lua [hunk 5]
```diff
@@ -2,7 +2,12 @@ return {
 	type = "function",
 	description = [[Returns an array-style table with a `string` split by a specified separator.
 The seperator is not part of the results. 
-By default the `sep == "%s"`, which will result in `str` getting split by whitespace characters (e.g. spaces and tabs).]],
+By default the `sep == "%s"`, which will result in `str` getting split by whitespace characters (e.g. spaces and tabs).
+
+!!! warning 
+	if `sep` is more than one character, then `str` will get split at each occurrence of any of the individual characters in `sep`.
+	For example, `string.split("1a2b3c4abc5", "abc")` will return `{"1", "2", "3", "4", "5"}` instead of `{"1a2b3c4", "5"}`.
+]],
```
Comment (C3pa) [inline autocomplete/definitions/global/string/split.lua:10]: I'll ask Null if this is a bug worth fixing

**Review by C3pa — COMMENTED** [review]
I am not sure what you note for the behavior of json functions is correct. IIRC, these functions return integer indices for pure array tables.

**Comment by herbertgithubaccount** [comment]
> I am not sure what you note for the behavior of json functions is correct. IIRC, these functions return integer indices for pure array tables.

oops. i double checked and you're right 😅. that's good to know. i'll fix it now

File: autocomplete/definitions/global/table/filter.lua (hunk 1 above)
Comment (herbertgithubaccount) [reply autocomplete/definitions/global/table/filter.lua:11]: good idea

**Review by C3pa — APPROVED** [review]

**Pull request merged**